src/pool.rs

```diff
@@ -302,13 +286,13 @@ impl ConnectionPool {
                 continue;
             }
 
-            allowed_attempts -= 1;
-
             // Don't attempt to connect to banned servers.
             if self.is_banned(address, shard, role) {
                 continue;
             }
 
+            allowed_attempts -= 1;
```

Comment:
If all the replicas are down, won't this cause the client to be stuck in an infinite loop in this function? I believe that's why I put `allowed_attempts -= 1` line above the `is_banned` check.

Reply:
Hmm. I think you are right.